src/components/forms/Combobox.tsx

```diff
@@ -883,6 +883,26 @@ export function ComboboxAsync<T = any>({
 
         if (inputOptions.value && inputOptions.value.toString().length > 0) {
           currentUrl.searchParams.set('with', inputOptions.value.toString());
+
+          if (currentUrl.searchParams.get('sort')) {
+            currentUrl.searchParams.delete('sort');
+          }
+        }
+
+        return currentUrl.href;
+      });
+    }
+
+    if (enableQuery && !inputOptions.value) {
+      setUrl((c) => {
+        const currentUrl = new URL(c);
+
+        if (currentUrl.searchParams.get('with')) {
+          currentUrl.searchParams.delete('with');
+        }
+
+        if (sortBy) {
+          currentUrl.searchParams.set('sort', sortBy);
         }
```

Comment:
When removing/clearing a value from the combobox, it is essential to ensure that the 'sort' parameter is included to obtain accurate results without the 'with' parameter (default list).